src/librustc_mir/shim.rs

```diff
@@ -540,7 +536,9 @@ impl<'a, 'tcx> CloneShimBuilder<'a, 'tcx> {
         // `let cloned = Clone::clone(rcvr[beg])`;
         // Goto #3 if ok, #5 if unwinding happens.
         let rcvr_field = rcvr.clone().index(beg);
-        let cloned = self.make_clone_call(ty, rcvr_field, BasicBlock::new(3), BasicBlock::new(5));
+        let cloned = self.make_place(Mutability::Not, ty);
```

Comment:
Please pass in `ret_field` from below as the destination (and rename it to `dest_elem` - also `rcvr_field` should be `src_elem`).